newIDE/app/src/EventsFunctionsExtensionEditor/OptionsEditorDialog/UsersAutocomplete.js

```diff
@@ -0,0 +1,149 @@
+//@flow
+import React from 'react';
+import { useState } from 'react';
+import { t } from '@lingui/macro';
+import SemiControlledMultiAutoComplete from '../../UI/SemiControlledMultiAutoComplete';
+import {
+  searchUserPublicProfilesByUsername,
+  type UserPublicProfile,
+  getUserPublicProfilesByIds,
+} from '../../Utils/GDevelopServices/User';
+
+import useForceUpdate from '../../Utils/UseForceUpdate';
+
+type Props = {|
+  eventsFunctionsExtension: gdEventsFunctionsExtension,
+|};
+
+export const UsersAutocomplete = (props: Props) => {
+  const forceUpdate = useForceUpdate();
+  const [authors, setAuthors] = React.useState([]);
+  const [authorInput, setAuthorInput] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [
+    completionUserPublicProfiles,
+    setCompletionUserPublicProfiles,
+  ] = React.useState<Array<UserPublicProfile>>([]);
+  const [authorsError, setAuthorsError] = React.useState(null);
+
+  // recalculate if the authorInput has changed
+  const searchUserPublicProfiles = React.useCallback(
+    async () => {
+      setAuthorsError(null);
+      if (!authorInput) {
+        setCompletionUserPublicProfiles([]);
+        return;
+      }
+      try {
+        setLoading(true);
+        const userPublicProfiles = await searchUserPublicProfilesByUsername(
+          authorInput
+        );
+        setCompletionUserPublicProfiles(userPublicProfiles);
+        setLoading(false);
+      } catch (err) {
+        setLoading(false);
+        setAuthorsError('Error while loading users');
+        console.log(err);
+      }
+    },
+    [authorInput]
+  );
+
+  // call every time the authorInput changes
+  React.useEffect(
+    () => {
+      searchUserPublicProfiles();
+    },
+    [searchUserPublicProfiles, authorInput]
+  );
+
+  // do only once, the first time it is loaded
+  const getUserPublicProfilesForAutocomplete = React.useCallback(
+    async () => {
+      setAuthorsError(null);
+      const authorIds = props.eventsFunctionsExtension
+        .getAuthorIds()
+        .toJSArray();
+      if (!authorIds.length) {
+        setAuthors([]);
+        return;
+      }
+      try {
+        setLoading(true);
+        const userPublicProfilesByIds = await getUserPublicProfilesByIds(
+          authorIds
+        );
+        setAuthors(
+          Object.keys(userPublicProfilesByIds).map(userId => {
+            const userPublicProfile: UserPublicProfile =
+              userPublicProfilesByIds[userId];
+            if (userPublicProfile.username && userPublicProfile.id) {
+              return {
+                text: userPublicProfile.username,
+                value: userPublicProfile.id,
+              };
+            }
+
+            return null;
```

Comment:
Question: should we display a "(no username)" in text (and the id in value)?

In theory this should never be shown to the user (because we'll have everyone to get a username, and you'll need one anyway to be selected).
In practice, I'm sure this will happen, so it's better to be clear in the UI rather than hiding the problem
(I prefer someone telling me "My username does not appear, help!" and find out that they got their username erased for some reason, than someone telling me "I disappeared from every extension, why did you remove me, I hate you!")

Reply:
Yeah I was under the impression it would never happen at all, but let's be safe and still display it!